File: 2021/kotlin/src/test/kotlin/nl/sanderp/aoc/aoc2021/Day02Test.kt
package nl.sanderp.aoc.aoc2021

import nl.sanderp.aoc.aoc2021.day02.*
import org.amshove.kluent.`should be equal to`
import kotlin.test.Test

class Day02Test {
    private val example = listOf(
        Forward(5),
        Down(5),
        Forward(8),
        Up(3),
        Down(8),
        Forward(2),
    )

    @Test
    fun `example part one`() {
        partOne(example) `should be equal to` 150
    }

    @Test
    fun `example part two`() {
        partTwo(example) `should be equal to` 900L
    }
}

File: 2022/scala/src/test/scala/AdventOfCode2022/Day02Test.scala
package AdventOfCode2022

import org.scalatest.flatspec.AnyFlatSpec

class Day02Test extends AnyFlatSpec {
  private val example = io.Source.fromResource("Day02.txt").getLines().toSeq

  "Part one" should "provide the correct answer for example input" in {
    assert(Day02.partOne(example) == 15)
  }

  "Part two" should "provide the correct answer for example input" in {
    assert(Day02.partTwo(example) == 12)
  }
}